src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,146 @@
+package duke.task;
+
+import duke.storage.Storage;
+import java.util.ArrayList;
+
+public class Task {
+    protected String name;
+    protected boolean isDone;
+    protected int id;
+    private static ArrayList<Task> listOfTask = new ArrayList<>(100);
+    private static int counter = 0;
+
```

Comment:
Since you define list of task as ArrayList, is it better to check array size directly?